### DISCUSSION TITLE
Ayuda con warnings del IDE

### DISCUSSION THREAD

**Discussion by clumsypunch** [IIC2133-PUC/2024-1#19] [Tarea 0]
El VScode me tira unos warnings raros pero me compila y funciona todo bien hasta esta parte, alguien que pueda ver el ultimo commit para debuggear porfavor? TG: @clumsypunch

**Comment by PabloNavarroY** [comment]
Por lo menos yo no puedo ver tu repositorio en específico. Si adjuntas capturas de pantalla pertinentes o das un ejemplo del tipo de advertencia que VSCode te da podremos ayudarte más rápido.

**Reply by clumsypunch** [reply]
Gracias Pablo, por confidencialidad prefiero esperar a @FelipeMunozMoraga o te puedo pasar las sc por dm y publicar la solucion aca mismo por si a alguien le ocurre algo parecido.

**Reply by FelipeMunozMoraga** [reply]
Hola, disculpen lo confundido pero por qué necesitan de mi para algo de este curso? No soy parte del equipo docente de EDD y rendí el curso el semestre pasado.
Quedo atento a cualquier cosa

**Reply by clumsypunch** [reply]
Disculpa Felipe fue un error, @PabloNavarroY cual es tu telegram? para enviarte pantallazos

**Reply by PabloNavarroY** [reply]
Te pido que analizes las advertencias que te da VSCode para que puedas identificar la raiz del problema y así poder hacer las preguntas pertinentes sin tener que mostrar tu código directamente. Puede que al final de esto ni siquiera necesites nuestra ayuda.

Te recomiendo encarecidamente que pases por este proceso porque esta una habilidad importante en la resolución de problemas.

**Comment by benjavicente** [comment]
¿Pudiste resolver esto @clumsypunch ?

Ojo que el editor depende de un _languaje server_. Es decir, un programa que analiza tu código para ver errores u otras cosas. Este  podría no estar funcionando por varias razones, pero la más típica es que falta, dado que el entorno está mal configurado, no existen las cosas necesarias para que pueda correr bien.

**Ahí depende de que tengas en tu computador, no de tu código**. Si mandas info de eso sería mucho más fácil identificar el problema.

Si estás en Windows, probablemente sea porque estás trabajando en el entorno de Windows y no el de WSL (ve la extensión que arregla eso). O en el raro caso en Mac, que falte descargar herramientas de desarrollador.

**Reply by clumsypunch** [reply]
Hola @benjavicente, gracias por tu respuesta. Estoy trabjando en `wsl` con las extensiones de c\c++ que recomienda instalar VScode, como tu dices puede ser algun problema de configuracion de estas o que tengo todo el programa en `void main()` 😋. En cualquier caso el codigo me compila sin errores, sin leaks y hasta me funciona en el test `T0/ship/parte-3-2-inc` pero en el ultimo se cae  `T0/ship/parte-3-hard` , estuve debuggendo desde el Domingo y todavi no pillo el bug. Adjunto sc, izquierda es el `correct-output.txt`. 
![Screenshot 2024-03-19 074815](https://github.com/IIC2133-PUC/2024-1/assets/101067836/b5a81e83-d9d1-400a-aeee-4828d75e5afc)
![Screenshot 2024-03-19 074825](https://github.com/IIC2133-PUC/2024-1/assets/101067836/b0acf6ad-7517-4c90-b94d-ff305d129b65)
![Screenshot 2024-03-19 074858](https://github.com/IIC2133-PUC/2024-1/assets/101067836/f9319ff5-7e76-40c4-8eba-635610556ce3)
![Screenshot 2024-03-19 074802](https://github.com/IIC2133-PUC/2024-1/assets/101067836/7344d5ad-f767-4076-96b9-e8c76d4872d6)

**Reply by benjavicente** [reply]
Por favor, si tienes una duda aparte, hasta en una pregunta/discusión aparte.

Ahí plz específica si se cae y termina el programa por un error o si te refieres que difiere el resultado.

**Reply by clumsypunch** [reply]
listo: https://github.com/orgs/IIC2133-PUC/discussions/30
